File: 0007_Reverse Integer/ReverseInteger.cs
class Solution {
    public int Reverse (int x) {
        long value = 0;
        bool negative = x < 0;
        long y = x;
        y = Math.Abs (y);

        while (y > 0) {
            value *= 10;
            value += y % 10;
            y /= 10;
        }

        if (value > int.MaxValue) {
            return 0;
        }

        int ret = (int) value;

        if (negative) {
            return 0 - ret;
        } else {
            return ret;
        }
    }
}

File: 0007_Reverse Integer/ReverseInteger.java
class Solution {
    public int reverse(int x) {
        if(x == 0) return 0;
        boolean negative = x < 0;
        x = Math.abs(x);
        
        long y = 0;
        while(x > 0)
        {
            y *= 10;
            y += (x % 10);
            x /= 10;
        }
        
        if(y > Integer.MAX_VALUE) return 0;
        int ans = (int)y;
        
        return negative ? -ans : ans;
    }
}